src/clinical/clinical-service.ts

```diff
@@ -219,27 +225,46 @@ export const getPaginatedClinicalData = async (
   query: PaginatedClinicalQuery,
 ) => {
   if (!programId) throw new Error('Missing programId!');
-  const start = new Date().getTime() / 1000;
 
   const allSchemasWithFields = await dictionaryManager.instance().getSchemasWithFields();
   // Get all donors + records for given entity
   const { donors, totalDonors } = await donorDao.findByPaginatedProgramId(programId, query);
 
   const taskToRun = WorkerTasks.ExtractEntityDataFromDonors;
-  const taskArgs = [donors as Donor[], totalDonors, allSchemasWithFields, query];
+  const taskArgs = [donors as Donor[], totalDonors, allSchemasWithFields, query.entityTypes, query];
 
   // Return paginated data
   const data = await runTaskInWorkerThread<{ clinicalEntities: ClinicalEntityData[] }>(
     taskToRun,
     taskArgs,
   );
 
-  const end = new Date().getTime() / 1000;
-  L.debug(`getPaginatedClinicalData took ${end - start}s`);
-
   return data;
 };
 
+export const getDonorEntityData = async (donorIds: number[]) => {
+  const allSchemasWithFields = await dictionaryManager.instance().getSchemasWithFields();
```

Comment:
no need to do anything about this one, just felt the need to acknowledge out loud how nervous it makes me to have these long chains without `?.` 😆

Reply:
long optional chaining makes me nervous too.
I actually really like how `dictionaryManager` does this - throws an error if there's no instance - so you could actually just catch the error and return ... empty array of donorIds in that case?

that node pattern of throw/catch actual errors vs just retuning nulls/undefined is used a lot in this repo and I think it works well.

Reply:
re `?` part of this.... `getSchemaWIthFields` will never actually be run on undefined, an error will be thrown instead

Reply:
welp! sounds like that requires a lot of "insider" knowledge on that `dictionaryManager`.

try+catch sure does sound desirable, but I guess my non-evident point is that this line is a bit "code golf"ey, and could use being broken into separate lines 🤷‍♂️

Reply:
don't need insider knowledge... the type system informs is of what the output will be. if we can trust the type system then we're safe here.

Reply:
oh fun! didn't realise TS was visible in PR reviews at Github. must get myself that browser extension!

Reply:
sarcasm acknowledged. agreed that the lack of introspection in the browser code review is limiting.

Reply:
to be fair, we could review PRs by testing them locally... like we used to back in the day.

Modern tooling has become a crutch for what could be more declarative; which proves hard to keep up in the review process. How do you suggest we improve that?